docs/kernel_providers.rst

```diff
@@ -0,0 +1,173 @@
+.. _kernel_providers:
+
+================
+Kernel Providers
+================
+
+
+Creating a kernel provider
+==========================
+
+By writing a kernel provider, you can extend how Jupyter applications discover
+and start kernels. For example, you could find kernels in an environment system
+like conda, or kernels on remote systems which you can access.
+
+To write a kernel provider, subclass
+:class:`KernelProviderBase <jupyter_kernel_mgmt.discovery.KernelProviderBase>`, giving your provider an ID
+and overriding two methods.
+
+.. class:: MyKernelProvider
+
+   .. attribute:: id
+
+      A short string identifying this provider. Cannot contain forward slash
+      (``/``).
+
+   .. method:: find_kernels()
+
+      Get the available kernel types this provider knows about.
+      Return an iterable of 2-tuples: (name, attributes).
+      *name* is a short string identifying the kernel type.
+      *attributes* is a dictionary with information to allow selecting a kernel.
+      This may also contain metadata describing other information pertaining to
+      the kernel's launch - parameters, environment, etc.
+
+   .. method:: launch(name, cwd=None, launch_params=None)
+    
+      Launches the kernel and returns a 2-tuple: (connection_info, kernel_manager).
+      *connection_info* is a dictionary consisting of the connection information 
+      relative to the launched kernel.
```

Comment:
relative -> relevant?

Reply:
I think relative may be more appropriate here since the connection-info is connected to the launched kernel.  I could also say _associated with_ or _pertaining to_.  Hmm. I like _pertaining to_ a bit more actually.